freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/migration/handler/DeleteCloudFormationHandler.java

```diff
@@ -0,0 +1,103 @@
+
+    @Inject
+    private AwsMigrationUtil awsMigrationUtil;
+
+    @Override
+    public Class<DeleteCloudFormationRequest> type() {
+        return DeleteCloudFormationRequest.class;
+    }
+
+    @Override
+    public void accept(Event<DeleteCloudFormationRequest> event) {
+        LOGGER.info("Delete the old cloud formation during the AWS migration");
+        DeleteCloudFormationRequest request = event.getData();
+        try {
+            CloudContext cloudContext = request.getCloudContext();
+            CloudCredential cloudCredential = request.getCloudCredential();
+            CloudStack cloudStack = request.getCloudStack();
+            AuthenticatedContext ac = awsAuthenticator.authenticate(cloudContext, cloudCredential);
+            Optional<CloudResource> cfCloudResource = resourceRetriever
+                    .findFirstByStatusAndTypeAndStack(CommonStatus.CREATED, ResourceType.CLOUDFORMATION_STACK, request.getResourceId());
+            boolean cloudFormationTemplateDeleted = false;
+            if (cfCloudResource.isPresent()) {
+                if (awsMigrationUtil.allInstancesDeletedFromCloudFormation(ac, cfCloudResource.get())) {
+                    terminateCfStackAndResource(cloudContext, cloudStack, ac, cfCloudResource.get());
+                    cloudFormationTemplateDeleted = true;
+                } else {
+                    LOGGER.info("Cannot delete the Cloudformation because it contains instances");
+                }
+            } else {
+                LOGGER.info("Cloud formation does not exist on AWS side, maybe deleted previously or by the customer");
+                cloudFormationTemplateDeleted = true;
+            }
+            DeleteCloudFormationResult result = new DeleteCloudFormationResult(request.getResourceId(), cloudFormationTemplateDeleted);
+            request.getResult().onNext(result);
+            eventBus.notify(result.selector(), new Event<>(event.getHeaders(), result));
+        } catch (Exception e) {
+            LOGGER.error("Cannot delete the AWS cloud formation template during the variant migration", e);
+            DeleteCloudFormationResult result = new DeleteCloudFormationResult(e.getMessage(), e, request.getResourceId());
+            request.getResult().onNext(result);
+            eventBus.notify(AWS_VARIANT_MIGRATION_FAILED_EVENT.event(), new Event<>(event.getHeaders(), result));
+        }
+    }
+
+    private void terminateCfStackAndResource(CloudContext cloudContext, CloudStack cloudStack, AuthenticatedContext ac,
+            CloudResource cfCloudResource) {
+        List<CloudResourceStatus> terminateResult = awsTerminateService.terminate(ac, cloudStack, List.of(cfCloudResource));
+        ResourcePersisted resourceDeletionResult = resourceNotifier.notifyDeletion(cfCloudResource, cloudContext);
+        LOGGER.debug("resource deletion result: {}", resourceDeletionResult);
+        LOGGER.info("Cloud formation terminated: {}", terminateResult);
```

Comment:
can we have a single line of log here? I don't think it makes much difference playing with the log level

Reply:
If it doesn't matter why don't log everything on info level?

Reply:
we could, as without debug logs we would be blind when investigating